### DISCUSSION TITLE
Unable to connect using SSL

### DISCUSSION THREAD

**Discussion by lig76** [felddy/foundryvtt-docker#922] [Q&A] [answered]
Hi,

I have setup Docker image on QNAP NAS device and it's running on port 30000 and I can connect w/o any issues. However I'd like to have it exposed externally, thus I need SSL. I'm trying to connect to the same internal IP address using https but I got "Unable to connect" error. I have set both:   "proxyPort": 443,  "proxySSL": true, but it doesn't help. Please share any advice with this issue, my ideas are already over :(

Best regards

**Comment by felddy** [comment, marked as answer]
Your question isn't specific to FoundryVTT on Docker which is what this project supports.  That said, take a look at the [official documentation](https://foundryvtt.com/kb/) for FoundryVTT as I think it covers your questions.  Specifically SSL, and what I gather to be port forwarding issues:

SSL:
- https://foundryvtt.com/article/ssl/

Port forwarding:
- https://foundryvtt.com/article/port-forwarding/

---

Since it is related, I'll also point out that I just published the first entry in the cookbook that shows how to securely deploy Foundry behind a Cloudflare tunnel.  If you have the motivation, I would highly recommend this implementation: 

- https://github.com/felddy/foundryvtt-docker/tree/develop/docs/cookbooks/cloudflare

```mermaid
  graph LR
    Users((Users)) -->|HTTPS| Edge
    subgraph Cloudflare
        Edge("Cloudflare Edge Server")
    end
    Cloudflare <-->|Tunnel| Cloudflared
    subgraph QNAP Host
        Cloudflared("Cloudflared Service")
        Foundry("FoundryVTT Service")
        Cloudflared -->|Local Traffic| Foundry
    end

```

**Comment by lig76** [comment]
Thank you !
